uber/models.py

```diff
@@ -1889,6 +1900,14 @@ def _track_changes(session, context, instances='deprecated'):
 
 
 def register_session_listeners():
+    """
+    IMPORTANT!!! Because we are locking our c.BADGE_LOCK at the start of this, all of these functions MUST NOT
```

Comment:
You probably want to inform any future coder of your very handy `@catch_all_exceptions` decorator here while telling them that this area of the code could deadlock the system. P: